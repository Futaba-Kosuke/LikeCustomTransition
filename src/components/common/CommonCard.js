/**
 * カード型のUIのベース部分を表示するコンポーネント
 */
import React from 'react';

//ReactNativeを使用したコンポーネントの呼び出し
import {
  StyleSheet,
  Image,
  Dimensions
} from 'react-native';

//NativeBaseを使用したコンポーネントの呼び出し
import {
  Content,
  Card,
  CardItem,
  Left,
  Body,
  Text,
  Button
} from 'native-base';

//デバイス幅の取得
var width = Dimensions.get('window').width;

//コンポーネントの内容を定義する ※ FunctionalComponent
//参考：ES2015(ES6)新構文：アロー関数(Arrow function)
const CommonCard = ({ album }) => {

  //取得した引数（オブジェクト：{ album }）を分割する
  const { id, title, category, image_url } = album;

  //表示する要素を返す
  return (
    <Content>
      <Card>
        <CardItem>
          <Body>
            <Text>{title}</Text>
            <Text note>写真番号：No.{id} ({category})</Text>
          </Body>
        </CardItem>
        <CardItem cardBody>
          <Image style={styles.imageStyle} source={{ uri: image_url }} />
        </CardItem>
        <CardItem content>
          <Text>このAPIから表示しているものはサンプルになります。</Text>
        </CardItem>
        <CardItem style={{ justifyContent: 'space-around' }}>
          <Button transparent>
            <Icon active name="thumbs-up" />
            <Text>12 Likes</Text>
          </Button>
          <Button transparent>
            <Icon active name="chatbubbles" />
            <Text>4 Comments</Text>
          </Button>
          <Text>2 days ago</Text>
        </CardItem>
      </Card>
    </Content>
  );
};

//このコンポーネントのStyle定義
const styles = {
  imageStyle: {
    width: width,
    marginLeft: 15,
    marginRight: 15,
    height: 300,
    flex: 1
  }
};

//このコンポーネントをインポート可能にする
export default CommonCard;
