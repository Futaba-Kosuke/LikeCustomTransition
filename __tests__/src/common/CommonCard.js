import 'react-native';
import React from 'react';
import CommonCard from '../../../src/components/common/CommonCard';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('共通コンポーネント(店舗表示用部分)のスナップショットテスト', () => {
  const shopDataMock = jest.fn();
  shopDataMock.mockReturnValue(
    {"shops":{
      "contents":[
        {
          "id":"1",
          "title":"大塚の油そば「麺屋 帝旺」",
          "category":"ラーメン・油そば","kcpy":"ピリ辛の特製ラー油がうまさの秘訣！",
          "detail":"少しピリ辛のラー油と食べ応えのある太麺が一度食べるとやみつきになる一品です。",
          "image_url":"https://otsuka-shop.s3-ap-northeast-1.amazonaws.com/shops/images/1/large.jpg"}
      ]
    }}
  );
  const tree = renderer.create(
    <CommonCard shop={shopDataMock().shops.contents[0]} onPress={ () => console.log("テスト") } />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
