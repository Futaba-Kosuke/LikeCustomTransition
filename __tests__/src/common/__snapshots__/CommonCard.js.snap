// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`共通コンポーネント(店舗表示用部分)のスナップショットテスト 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  onScroll={[Function]}
  resetScrollToCoords={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "transparent",
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#ccc",
          "borderRadius": 2,
          "borderWidth": 0.5,
          "elevation": 3,
          "flex": 1,
          "flexWrap": "wrap",
          "marginBottom": 5,
          "marginLeft": 2,
          "marginRight": 2,
          "marginTop": 5,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 1.5,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
          virtual={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 15,
                "lineHeight": 20,
              }
            }
            virtual={undefined}
          >
            大塚の油そば「麺屋 帝旺」
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            note={true}
            style={
              Object {
                "color": "#808080",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "200",
                "lineHeight": 20,
                "marginRight": 20,
                "marginTop": 5,
              }
            }
            virtual={undefined}
          >
            カテゴリー：
            ラーメン・油そば
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            note={true}
            style={
              Object {
                "color": "#808080",
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "200",
                "lineHeight": 20,
                "marginRight": 20,
              }
            }
            virtual={undefined}
          >
            ピリ辛の特製ラー油がうまさの秘訣！
          </Text>
        </View>
      </View>
      <View
        cardBody={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "paddingBottom": -5,
            "paddingLeft": -5,
            "paddingRight": -5,
            "paddingTop": -5,
          }
        }
        virtual={undefined}
      >
        <Image
          source={
            Object {
              "uri": "https://otsuka-shop.s3-ap-northeast-1.amazonaws.com/shops/images/1/large.jpg",
            }
          }
          style={
            Object {
              "flex": 1,
              "height": 240,
              "marginLeft": 15,
              "marginRight": 15,
            }
          }
        />
      </View>
      <View
        content={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#555",
              "fontFamily": "System",
              "fontSize": 14,
              "lineHeight": 22,
            }
          }
          virtual={undefined}
        >
          少しピリ辛のラー油と食べ応えのある太麺が一度食べるとやみつきになる一品です。
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 13,
            }
          }
          virtual={undefined}
        >
          更新日：2017/04/17
        </Text>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "transparent",
              "borderColor": "#007aff",
              "borderRadius": 5,
              "borderWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            active={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "color": "#007aff",
                  "fontSize": 24,
                  "marginLeft": 5,
                  "marginRight": 5,
                  "paddingTop": 2,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#007aff",
                "fontFamily": "System",
                "fontSize": 13,
                "lineHeight": 19,
                "marginLeft": 0,
                "marginRight": 0,
              }
            }
            virtual={undefined}
          >
            詳細情報を見る
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
